gettsim/benefits/alg2.py

```diff
@@ -221,50 +221,132 @@ def grossinc_alg2(household):
     )
 
 
-def e_anr_frei(household, params):
+def e_anr_frei_2005_01(household, params):
     """Calculate income not subject to transfer withdrawal for the household.
 
+    Legislation in force 2005-01-01 to 2005-09-30.
+
+    Determine the gross income that is not deducted. Withdrawal rates depend
+    on monthly earnings. § 30 SGB II."""
+
+    cols = ["m_wage", "ekanrefrei", "incometax", "soli", "svbeit"]
+    household.loc[:, cols] = household.groupby("pid")[cols].apply(
+        e_anr_frei_person_2005_01, params, params["a2eg3"]
+    )
+
+    return household
+
+
+def e_anr_frei_person_2005_01(person, params, a2eg3):
+    """Calculate income not subject to transfer withdrawal for each person.
+
+    Legislation in force 2005-01-01 to 2005-09-30."""
+
+    m_wage = person["m_wage"].iloc[0]
+
+    # Nettoquote
+    nq = alg2_2005_nq(person, params)
+
+    # Income not deducted
+    if m_wage <= params["a2eg1"]:
+        person["ekanrefrei"] = params["a2an1"] * nq * m_wage
+
+    elif params["a2eg1"] < m_wage <= params["a2eg2"]:
+        person["ekanrefrei"] = params["a2an1"] * nq * params["a2eg1"] + params[
+            "a2an2"
+        ] * nq * (m_wage - params["a2eg1"])
+
+    elif params["a2eg2"] < m_wage <= a2eg3:
+        person["ekanrefrei"] = (
+            params["a2an1"] * nq * params["a2eg1"]
+            + params["a2an2"] * nq * (params["a2eg2"] - params["a2eg1"])
+            + params["a2an3"] * nq * (m_wage - params["a2eg2"])
+        )
+
+    else:
+        person["ekanrefrei"] = (
+            params["a2an1"] * nq * params["a2eg1"]
+            + params["a2an2"] * nq * (params["a2eg2"] - params["a2eg1"])
+            + params["a2an3"] * nq * (a2eg3 - params["a2eg2"])
+        )
```

Comment:
Maybe write the ``* nq`` just once at the end of the formula.

Reply:
I think the current notation (following Knabe, 2006) is a bit more helpful in pointing out the weird mix of gross and net incomes in the initial ALG II withdrawal scheme. But really no strong opinion though.

Reply:
I agree with @mjbloemer . This is just closer to literature/law